genshin/models/genshin/chronicle/activities.py

```diff
@@ -43,6 +44,78 @@ class Activity(OldActivity[ModelT]):
     total_times: int = 1
 
 
+class DateTime(APIModel):
+    """DateTime model."""
+
+    year: int
+    month: int
+    day: int
+    hour: int
+    minute: int
+
+
+# ---------------------------------------------------------
+# Summer：
+
+
+class SummerStoryRecord(APIModel):
+    """Summer story record."""
+
+    finish_time: typing.Optional[DateTime] = Aliased("finish_time")
```

Comment:
This should be a `datetime.datetime` converted with a `@pydantic.validator`.

Reply:
![image](https://user-images.githubusercontent.com/72869479/179396716-edfbbd8c-1060-4ac7-bea5-289908399778.png)
We can handle this with `datetime.datetime` ?

Reply:
Also add tzinfo depending on what timezone hoyolab uses here.
```py
@pydantic.validator("finish_time", pre=True)
def __validate_time(cls, value: typing.Any) -> datetime.datetime:
    if isinstance(value, datetime.datetime):
         return value
    
    return datetime.datetime(value["year"], value["month"], value["day"], value["hour"], value["minute"], tzinfo=...)
```

Reply:
`tzinfo=datetime.timezone.utc` ? I don't have idea hoyolab uses which time zone, I think the data maybe base on game server time

Reply:
I was hoping you'd know when you might've finished your challenges. I'm banned so I cannot test it myself.

Reply:
I do testing by using datetime but it throw this error.Maybe something wrong?
![image](https://user-images.githubusercontent.com/72869479/179399205-fa095bb2-2125-4630-9d6c-73f72d61bc07.png)

Reply:
tested the finish time, it is fixed data